enginetest/queries/update_queries.go

```diff
@@ -730,6 +731,18 @@ var UpdateErrorTests = []QueryErrorTest{
 		Query:       `UPDATE people set height_inches = null where height_inches < 100`,
 		ExpectedErr: sql.ErrInsertIntoNonNullableProvidedNull,
 	},
+	{
+		Query:       `UPDATE people SET height_inches = IF(SUM(height_inches) % 2 = 0, 42, height_inches)`,
```

Comment:
do you think we've covered entrypoints here? or would it be useful to add some SELECT queries in `enginetest/queries.go`?

Reply:
Notably, subquery recursion can order rules for parent/child scopes in ways that aren't immediately intuitive. Testing combos of subquery aliases (relational subqueries) and subquery expressions (scalar subqueries) can flush out untested cases. Other scope-root nodes like Window, Union, CTE, recursive CTE are common offenders.

Reply:
I think that we're covered here, but my imagination for problematic queries may be lacking. Can you give an example of a query that has unintuitive rule ordering that you think might affect this?

Reply:
Queries with subqueries and subquery expressions in weird places sometimes break my first pass at new rules `select count((select * from (select x from xy) sq)) from xy;`

Reply:
can't think of anything that would break this PR right now, but trying to find edge cases is always a useful step